### ISSUE TITLE
False negative for RUF019 with `not` operator

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#12285]
RUF019 checks for things like `"key" in dct and dct["key"]` in boolean contexts, like an `if` condition or the argument to `bool`. It should also detect it as the operand of `not`.
```console
$ ruff --version
ruff 0.5.1
$ echo 'bool("key" in dct and dct["key"])' | ruff check --isolated --output-format=concise --select RUF019 /dev/stdin
/dev/stdin:1:6: RUF019 [*] Unnecessary key check before dictionary access
Found 1 error.
[*] 1 fixable with the `--fix` option.
$ echo 'not ("key" in dct and dct["key"])' | ruff check --isolated --output-format=concise --select RUF019 /dev/stdin
All checks passed!
```

**Comment by dhruvmanila** [comment]
This seems reasonable to me.

I think we'll need to visit the operand of a unary `not` operator using the `visit_boolean_test` which sets the correct flag on the semantic model:

https://github.com/astral-sh/ruff/blob/880c31d1642a6b4d967a20f9affe9772a3722b85/crates/ruff_linter/src/checkers/ast/mod.rs#L1796-L1804
